pkg/pipelineascode/match.go

```diff
@@ -248,3 +258,56 @@ func (p *PacRun) checkNeedUpdate(tmpl string) (string, bool) {
 	}
 	return "", false
 }
+
+func (p *PacRun) scopeTokenToListOfRipos(ctx context.Context, repo *v1alpha1.Repository) (string, error) {
+	var (
+		listRepos bool
+		token     string
+		err       error
+	)
+	listURLs := map[string]string{}
+	repoListToScopeToken := []string{}
+
+	// This is a Global config to provide list of repos to scope token
+	if p.run.Info.Pac.SecretGhAppTokenScopedExtraRepos != "" {
+		for _, configValue := range strings.Split(p.run.Info.Pac.SecretGhAppTokenScopedExtraRepos, ",") {
+			configValueS := strings.TrimSpace(configValue)
+			if configValueS == "" {
+				continue
+			}
+			repoListToScopeToken = append(repoListToScopeToken, configValueS)
+		}
+		listRepos = true
+	}
+	if len(repo.Spec.GithubAppTokenScopeRepos) != 0 {
+		ns := repo.Namespace
+		repoListInPerticularNamespace, err := p.run.Clients.PipelineAsCode.PipelinesascodeV1alpha1().Repositories(ns).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return "", err
+		}
+		for i := range repoListInPerticularNamespace.Items {
+			splitData := strings.Split(repoListInPerticularNamespace.Items[i].Spec.URL, "/")
+			listURLs[splitData[3]+"/"+splitData[4]] = splitData[3] + "/" + splitData[4]
+		}
+		for i := range repo.Spec.GithubAppTokenScopeRepos {
+			_, ok := listURLs[repo.Spec.GithubAppTokenScopeRepos[i]]
+			if !ok {
+				msg := fmt.Sprintf("repo %s does not exist in namespace %s", repo.Spec.GithubAppTokenScopeRepos[i], ns)
+				p.eventEmitter.EmitMessage(nil, zap.ErrorLevel, "RepoDoesnotExistInNamespace", msg)
+				return "", errors.New(msg)
+			}
+			repoListToScopeToken = append(repoListToScopeToken, repo.Spec.GithubAppTokenScopeRepos[i])
```

Comment:
can you add a info log that we are extending the token scope

Reply:
Added log and it looks like below now

```
Github token scope extended to [savitaashture/pacaccess savitaashture/pipeline]
```